src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -169,8 +169,48 @@ public void createRecord(String tableName, HttpEntity entity) throws IOException
       LOG.error("Error in creating a new record", e);
       throw new RuntimeException("Error in creating a new record");
     }
+
+    return systemID;
+  }
+
+  /**
+   * Fetches the System Id of a new Record.
+   *
+   * @param apiResponse API response after Creating a record
+   */
+  public String getSystemId(RestAPIResponse apiResponse) {
+
+    JsonObject jsonObject = gson.fromJson(apiResponse.getResponseBody(), JsonObject.class);
+    JsonObject result = (JsonObject) jsonObject.get(ServiceNowConstants.RESULT);
+
+    return result.get(ServiceNowConstants.SYSTEM_ID).getAsString();
   }
 
+  /**
+   * Return a record from ServiceNow application.
+   *
+   * @param tableName The ServiceNow table name
+   * @param query     The query
+   */
+  public Map<String, String> getRecordFromServiceNowTable(String tableName, String query)
```

Comment:
@VisibleForTesting

Reply:
I am actually using this method in the e2e-test package.